src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java

```diff
@@ -49,6 +55,10 @@ public class QueryServiceConnection implements Connection {
     private boolean enableStreamFlow = false;
     private String tenantUrl;
     private QueryEngineEnum queryEngineEnum;
+    private Cache<MetadataCacheKey, String> metaDataCache;
+    private boolean addMetaDataInterceptor;
+    @Getter
+    private final int metaDataCacheDurationInMs;
```

Comment:
Client may pass in the value through properties right? How do I access this variable in interceptor?

Reply:
@soaggarwal You meant to take the properties from `getClientInfo`?

Reply:
@KaviarasuSakthivadivel : Instead of setting this property in Connection, you can use getClientInfo and look up the property value when the MetadataInterceptor is initialized